llm_studio/src/utils/modeling_utils.py

```diff
@@ -58,21 +58,24 @@ def check_disk_space(model: torch.nn.Module, path: str, use_deepspeed: bool = Fa
 
     model_size_in_bytes = 0
     for param in model.parameters():
+        n_params = param.ds_numel if hasattr(param, "ds_numel") else param.numel()
         if param.data.dtype in [torch.int8, torch.uint8]:
-            model_size_in_bytes += param.numel() * 1
+            model_size_in_bytes += n_params * 1
         elif param.data.dtype in [torch.float16, torch.bfloat16]:
-            model_size_in_bytes += param.numel() * 2
+            model_size_in_bytes += n_params * 2
         elif param.data.dtype == torch.float32:
-            model_size_in_bytes += param.numel() * 4
+            model_size_in_bytes += n_params * 4
         else:
             # If the data type is not supported, calculate it as float32.
-            model_size_in_bytes += param.numel() * 4
+            model_size_in_bytes += n_params * 4
             logger.warning(f"Unsupported data type: {param.data.dtype}")
 
-    if use_deepspeed:
-        model_size_in_bytes *= 2  # need double space for converting deepspeed engine.
```

Comment:
by saving pytorch checkpoint directly for deepspeed, we don't need to `x2` for space check now.